packages/contracts-bedrock/src/cannon/PreimageOracle.sol

```diff
@@ -183,6 +183,90 @@ contract PreimageOracle is IPreimageOracle {
         preimageLengths[key] = size;
     }
 
+    /// @inheritdoc IPreimageOracle
+    function loadBlobPreimagePart(
+        uint256 _z,
+        uint256 _y,
+        bytes calldata _commitment,
+        bytes calldata _proof,
+        uint256 _partOffset
+    )
+        external
+    {
+        bytes32 key;
+        bytes32 part;
+        assembly {
+            // Compute the versioned hash. The SHA2 hash of the 48 byte commitment is masked with the version byte,
+            // which is currently 1. https://eips.ethereum.org/EIPS/eip-4844#parameters
+            // SAFETY: We're only reading 48 bytes from `_commitment` into scratch space, so we're not reading into the
+            //         free memory ptr region.
+            calldatacopy(0x00, _commitment.offset, 0x30)
+            let success := staticcall(gas(), 0x02, 0x00, 0x30, 0x00, 0x20)
```

Comment:
I would add to the safety comment here that it's ok to place the commitment into the scratch space even though hashing is done (which could use scratch space) since it is used by the hash function. So it won't matter if it's overwritten and the scratch space is not expected to be clean so it won't dirty any hash function operations.